src/test/java/com/metamx/metrics/MonitorUtilsTest.java

```diff
@@ -0,0 +1,28 @@
+package com.metamx.metrics;
+
+import com.google.common.collect.ImmutableMap;
+import com.metamx.emitter.service.ServiceMetricEvent;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MonitorUtilsTest
+{
+  @Test
+  public void TestAddDimensionsToBuilder()
+  {
+    ServiceMetricEvent.Builder builder = new ServiceMetricEvent.Builder();
+    Map<String, String[]> dimensions = ImmutableMap.of(
+        "dim1", new String[] { "value1" },
+        "dim2", new String[] { "value2.1", "value2.2" }
+    );
+
+    MonitorUtils.addDimensionsToBuilder(builder, dimensions);
+
+    Assert.assertEquals(((List) builder.getDimension("dim1")).get(0), "value1");
+
+    Assert.assertEquals(((List) builder.getDimension("dim2")).get(0), "value2.1");
+    Assert.assertEquals(((List) builder.getDimension("dim2")).get(1), "value2.2");
```

Comment:
then just compare two lists instead, you can create the other list to compare against using `ImmutableList.of("value2.1", "value2.2")`